src/envoy/auth/jwt.cc

```diff
@@ -90,19 +102,49 @@ const uint8_t *CastToUChar(const std::string &str) {
   return reinterpret_cast<const uint8_t *>(str.c_str());
 }
 
-bssl::UniquePtr<EVP_PKEY> EvpPkeyFromStr(const std::string &pkey_pem) {
-  std::string pkey_der = Base64::decode(pkey_pem);
-  bssl::UniquePtr<RSA> rsa(
-      RSA_public_key_from_bytes(CastToUChar(pkey_der), pkey_der.length()));
+bssl::UniquePtr<EVP_PKEY> EvpPkeyFromRsa(const bssl::UniquePtr<RSA> &rsa) {
   if (!rsa) {
     return nullptr;
   }
   bssl::UniquePtr<EVP_PKEY> key(EVP_PKEY_new());
   EVP_PKEY_set1_RSA(key.get(), rsa.get());
-
   return key;
 }
 
+bssl::UniquePtr<EVP_PKEY> EvpPkeyFromStr(const std::string &pkey_pem) {
+  std::string pkey_der = Base64::decode(pkey_pem);
+  return EvpPkeyFromRsa(bssl::UniquePtr<RSA>(
+      RSA_public_key_from_bytes(CastToUChar(pkey_der), pkey_der.length())));
+}
+
+BIGNUM *BigNumFromBase64UrlString(const std::string &s) {
```

Comment:
Using UniquePtr<BIGNUM> doesn't work because it causes that rsa->n and rsa->e are deleted before signature validation. Any idea to resolve this?